src/components/card/CardSidebarTabAttachments.vue

```diff
@@ -22,30 +22,99 @@
 
 <template>
 	<div>
-		{{ card.attachments }}
+		<li v-for="attachment in attachments" :key="attachment.id" class="attachment">
+			<a class="fileicon" :style="mimetypeForAttachment(attachment)" />
+			<div class="details" :href="attachmentUrl(attachment)">
+				<div class="filename">
+					<span class="basename">{{ attachment.data }}</span>
+				</div>
+				<span class="filesize">{{ attachment.extendedData.filesize }}</span>
+				<span class="filedate">{{ attachment.createdAt }}</span>
+				<span class="filedate">{{ attachment.createdBy }}</span>
+			</div>
+			<Actions>
+				<ActionButton icon="icon-delete" @click="deleteAttachment(attachment)">
+					{{ t('deck', 'Delete Attachment') }}
+				</ActionButton>
+
+				<ActionButton v-if="attachment.extendedData.mimetype.includes('image')" icon="icon-picture" @click="coverToggle(attachment)">
+					{{ t('deck', 'Use as cover') }}
+				</ActionButton>
+			</Actions>
+		</li>
 		<button class="icon-upload" @click="clickAddNewAttachmment()">
 			{{ t('deck', 'Upload attachment') }}
 		</button>
+		<input ref="localAttachments"
+			type="file"
+			multiple
+			style="display: none;"
+			@change="onLocalAttachmentSelected">
 	</div>
 </template>
 
 <script>
+import { Actions, ActionButton } from '@nextcloud/vue'
+
 export default {
 	name: 'CardSidebarTabAttachments',
+	components: {
+		Actions,
+		ActionButton,
+	},
 	props: {
 		card: {
 			type: Object,
 			default: null,
 		},
 	},
+	data() {
+		return {
+		}
+	},
+	computed: {
+		attachments() {
+		  return this.$store.getters.attachmentsByCard(this.card.id)
+		},
+
+	},
+	created: function() {
+		this.$store.dispatch('getAllAttachments', this.card.id)
+	},
 	methods: {
 		clickAddNewAttachmment() {
-
+			this.$refs.localAttachments.click()
+		},
+		onLocalAttachmentSelected(e) {
+			const bodyFormData = new FormData()
+			bodyFormData.append('cardId', this.card.id)
+			bodyFormData.append('type', 'deck_file')
+			bodyFormData.append('file', e.target.files[0])
```

Comment:
Related to the previous multiple comment, this just uploads the first file. Either the backend needs to support uploading multiple files in one API call or you need to do one for each file in the frontend. If done in the frontend we need some proper progress display for this and queue them somehow.